components/ai_chat/core/browser/ai_chat_metrics.cc

```diff
@@ -351,6 +519,81 @@ void AIChatMetrics::ReportChatCounts() {
   // TODO(djandries): remove the following report when Nebula experiment is over
   p3a_utils::RecordToHistogramBucket(kChatCountNebulaHistogramName,
                                      kChatCountBuckets, chat_count);
+
+  uint64_t max_chat_duration =
+      chat_durations_storage_.GetHighestValueInPeriod();
+  p3a_utils::RecordToHistogramBucket(kMaxChatDurationHistogramName,
+                                     kMaxChatDurationBuckets,
+                                     max_chat_duration);
+
+  uint64_t chat_with_history_count =
+      chat_with_history_count_storage_.GetWeeklySum();
+  int history_percentage = static_cast<int>(std::ceil(
+      chat_count > 0
+          ? (static_cast<double>(chat_with_history_count) / chat_count) * 100.0
+          : 0.0));
+
+  p3a_utils::RecordToHistogramBucket(kChatHistoryUsageHistogramName,
+                                     kChatHistoryUsageBuckets,
+                                     history_percentage);
+}
+
+void AIChatMetrics::RecordContextSource(ConversationHandlerForMetrics* handler,
+                                        mojom::ConversationTurnPtr& entry) {
+  ContextSource context = ContextSource::kTextInputWithoutPage;
+  if (prompted_via_omnibox_) {
+    context = ContextSource::kOmniboxInput;
+  } else if (entry->action_type == mojom::ActionType::SUMMARIZE_PAGE ||
+             entry->action_type == mojom::ActionType::SUMMARIZE_VIDEO) {
+    context = ContextSource::kPageSummary;
+  } else if (handler->GetConversationHistorySize() == 1 &&
+             entry->action_type == mojom::ActionType::CONVERSATION_STARTER) {
```

Comment:
Does one of the unit tests verify that the entry has already been added to the history when the first entry is submitted and this function is called? I ask in case this changes and this function ends up being called where `GetConversationHistorySize() == 0`, we'd want a test that fails.